### PR TITLE
Adding Semgrep Rules for SSRF in Java

### PR DESCRIPTION
**Pull request by salecharohit**
Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries#editor--validation-policy).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries#conversion-rules).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as [TEXT](URL)
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://travis-ci.org/OWASP/CheatSheetSeries/pull_requests).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #457 

Thank you again for your contribution :smiley:

### CODE REVIEW

**Comment by mackowski** [comment]
@salecharohit thank you for PR! I will review it this week as I want to review rules also :) But it looks solid on first glance

**Comment by salecharohit** [comment]
> @salecharohit thank you for PR! I will review it this week as I want to review rules also :) But it looks solid on first glance

Sure. I have a PR pending at Semgrep side as well. It'll be reviewed soon and will be published too.

**Comment by salecharohit** [comment]
> @salecharohit thank you for PR! I will review it this week as I want to review rules also :) But it looks solid on first glance

Sure. I have a PR pending at Semgrep side as well. It'll be reviewed soon and will be published too.

File: cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md [hunk 1]
```diff
@@ -378,6 +378,12 @@ In cloud environments SSRF is often used to access and steal credentials and acc
 
 To leverage this protection migrate to IMDSv2 and disable old IMDSv1. Check out [AWS documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html) for more details.
 
+## Semgrep Rules
+
+[Semgrep](https://semgrep.dev/) is a command-line tool for offline static analysis. Use pre-built or custom rules to enforce code and security standards in your codebase.
+Checkout the Semgrep rule for SSRF to identify/investigate for SSRF vulnerabilities in Java
+[https://semgrep.dev/salecharohit:owasp_java_ssrf](https://semgrep.dev/salecharohit:owasp_java_ssrf)
```
Comment (mackowski) [inline cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md:385]: I see that this link is to semgrep.dev/salecharohit:owsap_java_ssrf will it change after they merge your PR? Should we wait with this PR until they merge it master?

**Comment by mackowski** [comment]
@salecharohit any updates on this? Is your PR at Semgrep side merged?

**Comment by salecharohit** [comment]
> @salecharohit any updates on this? Is your PR at Semgrep side merged?

Yep. Here is the files in [github](https://github.com/returntocorp/semgrep-rules/tree/develop/contrib/owasp/java)
The Semgrep URL for testing these rules are [here](https://semgrep.dev/3eee) 

Let me know if any other details are required.

**Comment by mackowski** [comment]
@salecharohit apologies for late replay but I was on a short break last week :) 
I think that it will be better to link to GH https://github.com/returntocorp/semgrep-rules/blob/develop/contrib/owasp/java/ssrf.yaml in line 385 what do you think?

**Comment by salecharohit** [comment]
> @salecharohit apologies for late replay but I was on a short break last week :)
> I think that it will be better to link to GH https://github.com/returntocorp/semgrep-rules/blob/develop/contrib/owasp/java/ssrf.yaml in line 385 what do you think?

Yep sure absolutely no problem. The only reason why I added the link to semgrep.live is because people can quickly run the rule and see the results.

**Comment by salecharohit** [comment]
Hi @mackowski awaiting your decision as my XXE rules are also ready to integrate. I'd rather wait to send a PR for XXE once the SSRF is approved.

**Review by mackowski — APPROVED** [review]
Looks good

**Comment by mackowski** [comment]
@salecharohit go ahead and create PR for XXE - thank you for your contribution and apologies for long reply.

**Pull request merged**